custom_components/teslafi/coordinator.py

```diff
@@ -88,15 +88,25 @@ def _schedule_refresh(self) -> None:
 
     async def _refresh(self) -> TeslaFiVehicle:
         """Refresh"""
-        current = await self._client.last_good()
-        LOGGER.debug("Last good: %s", current)
-
+        was_sleeping = self._vehicle.is_sleeping
+        current = await self._client.current_data()
+        LOGGER.debug("Current: %s", current)
+        
         self._infer_charge_session(prev=self.data, current=current)
+        
+        if current.is_sleeping and not was_sleeping:
+            LOGGER.debug("Car is now sleeping, fetching last good data")
+            last_good = await self._client.last_good()
+            LOGGER.debug("Last good: %s", last_good)
+            # Populating last good data as current will have numerous empty fields when car is sleeping
+            self._vehicle.update_non_empty(last_good)
+            
+            assert last_good.vin
 
         self._vehicle.update_non_empty(current)
```

Comment:
One thing that might come up is if the future polling intervals might accidentally overwrite the `last_good` values, with `current` values that are empty due to sleeping - since it will only fill in the `last_good` data on the first transition from awake->asleep.

But I'll merge this as it is and test it on my own car to see how it handles that. Assuming nothing major, I'll get this released in the next few days.